### PR TITLE
Adapt for dask changes in shuffle

### PR DESCRIPTION
**Pull request by phofl**
CI coverage for the released version is something we should have added already

### CODE REVIEW

**Comment by phofl** [comment]
cc @mrocklin for a quick review maybe?

For context: Rick and I agreed that we want to support the latest released version and main for the time being. We will remove this special casing as soon as the next dask release is available and bump the minimum version in the next dask-expr release.

File: ci/environment_released.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: dask-expr
+channels:
+  - conda-forge
+dependencies:
+  - pytest
+  - pytest-cov
+  - dask  # overridden by git tip below
```
Comment (mrocklin) [inline ci/environment_released.yml:7]:
```suggestion
  - dask
```

File: dask_expr/_merge.py [hunk 2]
```diff
@@ -515,7 +515,10 @@ def assign_index_merge_transfer(
             index = partitioning_index(index, npartitions)
         df[name] = index
         meta[name] = 0
-        return merge_transfer(df, id, input_partition, npartitions, meta, parts_out)
+        disk = [] if not DASK_GT_20231000 else (True,)
```
Comment (mrocklin) [inline dask_expr/_merge.py:518]:
```suggestion
        disk = () if not DASK_GT_20231000 else (True,)
```

Just me being pedantic.   My apologies.

**Review by mrocklin — APPROVED** [review]
Sure.  Seems fine to me.  Two (very trivial) suggestions below.

**Pull request merged**